File: src/14. Longest Common Prefix/Solution.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if len(strs) == 0:
            return ""
        res = strs[0]
        for i in strs[1:]:
            while len(res)>0 and res not in i[:len(res)]:
                res = res[:len(res)-1]
        return res

File: src/14. Longest Common Prefix/Solution.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.size()==0) return "";
        for(int i=0;i<strs[0].length();i++){
            for(int j=0;j<strs.size();j++){
                if(strs[0][i]!=strs[j][i]||i>strs[j].length()){
                    return strs[0].substr(0,i);
                }
            }
        }
        return strs[0];
    }
};
